storage/grpc_client.go

```diff
@@ -425,8 +431,23 @@ func (c *grpcStorageClient) ListBucketACLs(ctx context.Context, bucket string, o
 	}
 	return attrs.ACL, nil
 }
+
 func (c *grpcStorageClient) UpdateBucketACL(ctx context.Context, bucket string, entity ACLEntity, role ACLRole, opts ...storageOption) (*ACLRule, error) {
-	return nil, errMethodNotSupported
+	// Get the existing acl from BucketAttrs
+	attrs, err := c.GetBucket(ctx, bucket, nil, opts...)
+	if err != nil {
+		return nil, err
+	}
+	var acl []ACLRule
+	acl = append(attrs.ACL, ACLRule{Entity: entity, Role: role})
+	// Set acl in BucketAttrsToUpdate
+	uattrs := &BucketAttrsToUpdate{acl: acl}
```

Comment:
As discussed in Carlos's doc, we need to set an ifMetagenerationMatch precondition here to prevent a data race. We can discuss offline.

Reply:
Added the MetagenerationMatch precondition. I know we talked about it, forgot to add it in >< Thanks for catching this!